### PR TITLE
[compiler] Add Columnar STypes/ETypes

### PR DESCRIPTION
**Pull request by chrisvittal**

### CODE REVIEW

File: hail/src/main/scala/is/hail/types/physical/stypes/concrete/SStructOfArrays.scala [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    )
+  }
+}
+
+class SStructOfArraysSettable(
+  val st: SStructOfArrays,
+  val length: Settable[Int],
+  val missing: Option[Settable[Long]],
+  val fields: IndexedSeq[SIndexableValue with SSettable]
+) extends SIndexableValue with SSettable {
```
Comment (chrisvittal) [inline hail/src/main/scala/is/hail/types/physical/stypes/concrete/SStructOfArrays.scala:75]: We're going to want to add specialized methods for accessing only one field in a tight loop.

File: hail/src/main/scala/is/hail/types/physical/stypes/concrete/SStructOfArrays.scala [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+      elementsRequired,
+      structType.types.zip(fields).map { case (v, f) => f.castRename(TArray(v)).asInstanceOf }
+    )
+  }
+}
+
+class SStructOfArraysSettable(
+  val st: SStructOfArrays,
+  val length: Settable[Int],
+  val missing: Option[Settable[Long]],
```
Comment (chrisvittal) [inline hail/src/main/scala/is/hail/types/physical/stypes/concrete/SStructOfArrays.scala:73]: If the elements are optional and the field is optional, I intend to have the field missingness be an `OR` of these missing bits and the array's missing bits. That way accessing a single field can be entirely local to that field.

**Review by chrisvittal — COMMENTED** [review]
Some initial comments.

**Comment by danking** [comment]
I'm closing this because its so old. We can start a new PR when we are ready to revisit this.

**Pull request closed without merging**